sdlib/d/gc/extent.d

```diff
@@ -324,8 +342,8 @@ public:
 			return;
 		}
 
-		// Encode freespace and write it to the last byte (or two bytes) of alloc.
-		writePackedFreeSpace(cast(ushort*) (address + slotSize - 2),
+		// Encode freespace and write it to the last usable byte (or two bytes):
+		writePackedFreeSpace(getFreeSpacePosition(index),
```

Comment:
Get free space position: with 48 bits of address space you have 16bits free IN the pointer.

Reply:
No. Either way, the fact we disagree is a strong sign a test is needed.